polaris/polaris/sep10auth/views.py

```diff
@@ -29,150 +35,179 @@
-    can then sign it using their private key and hit `POST /auth` to receive a JSON web
-    token. That token can be used to authenticate calls to the other SEP 24 endpoints.
-    """
-    if request.method == "POST":
-        return _post_auth(request)
-    return _get_auth(request)
+    ################
+    # POST functions
+    ################
+    def post(self, request: Request, *args, **kwargs) -> JsonResponse:
+        try:
+            envelope_xdr = self._get_transaction_xdr(request)
+            self._validate_challenge_xdr(envelope_xdr)
+        except ValueError as e:
+            return JsonResponse({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return JsonResponse({"token": self._generate_jwt(request, envelope_xdr)})
+
+    @staticmethod
+    def _validate_challenge_xdr(envelope_xdr: str):
+        """
+        Validate the provided TransactionEnvelope XDR (base64 string).
+
+        If the source account of the challenge transaction exists, verify the weight
+        of the signers on the challenge are signers for the account and the medium
+        threshold on the account is met by those signers.
+
+        If the source account does not exist, verify that the keypair used as the
+        source for the challenge transaction has signed the challenge. This is
+        sufficient because newly created accounts have their own keypair as signer
+        with a weight greater than the default thresholds.
+        """
+        server_key = settings.SIGNING_KEY
+        net = settings.STELLAR_NETWORK_PASSPHRASE
+
+        logger.info("Validating challenge transaction")
+        try:
+            tx_envelope, account_id = read_challenge_transaction(
+                envelope_xdr, server_key, net
+            )
+        except InvalidSep10ChallengeError as e:
+            err_msg = f"Error while validating challenge: {str(e)}"
+            logger.error(err_msg)
+            raise ValueError(err_msg)
+
+        try:
+            account = settings.HORIZON_SERVER.load_account(account_id)
+        except NotFoundError:
+            logger.warning(
+                "Account does not exist, using client's master key to verify"
+            )
+            try:
+                verify_challenge_transaction_signed_by_client(
+                    envelope_xdr, server_key, net
+                )
+            except InvalidSep10ChallengeError as e:
+                logger.info(
+                    f"Missing or invalid signature(s) for {account_id}: {str(e)})"
+                )
+                raise ValueError(str(e))
+            else:
+                logger.info("Challenge verified using client's master key")
+                return
+
+        signers = account.load_ed25519_public_key_signers()
+        threshold = account.thresholds.med_threshold
+        try:
+            signers_found = verify_challenge_transaction_threshold(
```

Comment:
Is there a plan to do something with the `signers_found` captured from the return that got missed, or we're just putting it in the variable to remind readers that they're available?

Reply:
Its used in the log statement, but there is no plan to use it functionally. I'm not particularly attached to having that log statement though so if you think its not needed I can remove it.

Reply:
Ah, no worries. Sorry I missed the log statement. It's fine to leave, I just wasn't sure if there was something incomplete, but it looks good to me.